bundles/org.smarthomej.binding.tuya/README.md

```diff
@@ -0,0 +1,123 @@
+
+## Thing Configuration
+
+### `project`
+
+First create and link a Tuya Develop Account:
+
+- Go to `iot.tuya.com` (the Tuya developer portal) and create an account. 
+You can choose any credentials (email/password) you like (it is not necessary that they are the same as in the app). 
+After confirming your account, log in to your new account. 
+- On the left navigation bar, select "Cloud", then "Create new Cloud project" (upper right corner).
+Enter a name (e.g. "My Smarthome"), select "Smart Home" for "Industry" and "Development Method".
+For security reasons, select only the "Data Center" that your app is connected to (you can change that later if you select the wrong one). 
+Select "IoT Core", "Authorization" and "Device Status Notification" as APIs.
+- You should be redirected to the "Overview" tab of your project. 
+Write down (or copy) "Access ID/Client ID" and "Access Secret/Client Secret" (ypu can always look it up in your account).
+- In the upper menu bar, select the "Devices" tab, then go to "Link Tuya App Account" and link you App account.
+
+Add a `project` and enter your credentials (`username`/`password`, from the app - not your cloud account!) and the cloud project credentials (`accessId`/`accessSecret`).
+The `countryCode` is the international dial prefix of the country you registered your app in (e.g. `49` for Germany or `43` for Austria).
+Depending on the app you use, set `schema` to `tuyaSmart` (for the Tuya Smart app) or `smartLife` (for the Smart Life app).
+The `datacenter` needs to be set to the same value as in your IoT project.
+
+The thing should come online immediately.
+
+If the thing does not come online, check 
+
+- if you entered the correct country code (check in the App if you accidentally choose a wrong country)
+- check if you selected the correct "Data Center" in your cloud project (you can select more than one for testing).
+
+### `tuyaDevice`
+
+The best way to configure a `tuyaDevice` is using the discovery service.
+
+The mandatory parameters are `deviceId`, `productId` and `localKey`.
+The `deviceId` is used to identify the device, the `productId` identifies the type of the device and the `localKey` is a kind of password for access control.
+These parameters are set during discovery.
+If you want to manually configure the device, you can also read those values from the cloud project above.
+
+For line powered device on the same subnet `ip` address and `protocol` version are automatically detected.
+Tuya devices announce their presence via UDP broadcast packets, which is usually not available in other subnets.
+Battery powered devices do not announce their presence at all.
+There is no clear rule how to determine if a device has protocol 3.3 or 3.1.
+It is recommended to start with 3.3 and watch the log file if it that works and use 3.1 otherwise.
+
+*Note:* Support for protocol 3.1 is considered experimental.
```

Comment:
```suggestion
*Note:* Support for protocol 3.1 is considered experimental.
In case something is not working, please open an issue on [GitHub](https://github.com/smarthomej/addons/issues) and add TRACE level logs.
```